cockpit/src/subscriptions-view.jsx

```diff
@@ -177,128 +212,86 @@ class SubscriptionStatus extends React.Component {
             this.props.unregister();
         e.stopPropagation();
     }
+
     render() {
-        let errorMessage;
-        if (this.props.error) {
-            errorMessage = (
-                <DismissableError dismissError={this.props.dismissError}
-                                  severity={this.props.error.severity}>
-                    {this.props.error.msg.toString()}
-                </DismissableError>
+        let syspurpose = null;
+        const p = this.props.syspurpose;
+        if (p["service_level_agreement"] || p["usage"] || p["role"] || p["addons"]) {
+            syspurpose = (
+                <Card id="syspurpose" key="syspurpose" className="ct-card-info">
+                    <CardHeader>
+                        <CardHeaderMain>
+                            <Text className="purpose-header" component={TextVariants.h2}>{_("System purpose")}</Text>
+                            {"(" + this.props.syspurpose_status + ")"}
```

Comment:
> the brackets must be translatable (they could be written with different unicode characters in other languages), so the whole layout (e.g. in French there are spaces around certain punctuation, so you need a space after the opening bracket and a space before the closing one).

Hmm, this is not a sentence, so I would not worry about grammar in general - it is just to distinguish it from the title. Fair point about languages that might be using different characters - can you name such language? I checked Weblate and for example Chinese is using normal parentheses as well. 

I find it somewhat counterproductive to translate `(%1)` as I think that every language would just keep it as it is. We had bunch of such strings where we would be translating stuff like `@` and so on but in general we always stopped doing that as it just puts unnecessary work on translators shoulders.

Reply:
> Hmm, this is not a sentence, so I would not worry about grammar in general

Well, it's a UI string, shown to the user.

> I find it somewhat counterproductive to translate (%1) as I think that every language would just keep it as it is.

This is not about how many languages would change it, but rather that this is a string/word puzzle: untranslated parts of UI strings are mixed together with translatable bits, and translators have no way to know how the smaller bits fit together, and not even the possibility to change how they fit together. In general, it's always a better idea to translate single sentences as such.

See also a bit of explanation from the KDE project: https://techbase.kde.org/Development/Tutorials/Localization/i18n_Mistakes#Pitfall_.232:_Word_Puzzles

Reply:
> untranslated parts of UI strings are mixed together with translatable bits, and translators have no way to know how the smaller bits fit together, and not even the possibility to change how they fit together. In general, it's always a better idea to translate single sentences as such.

If you notice, nothing from this part is being translated. `this.props.syspurpose_status` comes from API (there are ways how this can be translated if API provides untranslated string). So the situation is
```
<h2>Translated string</h2>
<span>(untraslated string)</span>
```
thus translators won't be able to "change how they fit together" as these are two unrelated elements. Your approach would just let them translate `(%1)` additionally. If the whole title would be one string as `System purpose (%1)` then I would of course agree that it should not be done this way.
But if you strongly believe that someone may need to change translation of parentheses (or space in them) I can change it, in the end, it is not me, who will be translating it :)